examples/installation_on_windows/README.md

```diff
@@ -35,7 +35,7 @@ If everything goes right, you should have this terminal. The command line starts
 
 
 ## 2 Configure the WSL connection
-Back to the installation of sllm, use `free -h` to check the memory status. After testing, the minimum setting is 22GB Mem and 8GB Swap as shown below. If it didn't meet this requirement, you need to be set manually in the file `.wslconfig` which should locate in `C:\Users\your_username\`. If not, just create it. Within the file, copy and paste the setting below. WSL2's VM allocates a certain amount of memory, and Swap space is managed within that VM's memory allocation. This means the Swap memory potentially consumes the RAM in WSL connection. Therefore, the memory needs to be much bigger than Swap to run the sllm successfully. After testing on different WSL config settings, the minimum memory and swap required is listed below.
+Back to the installation of sllm, use `free -h` to check the memory status. After testing, the minimum setting is 22GB Mem and 8GB Swap as shown below. If it didn't meet this requirement, you need to be set manually in the file `.wslconfig` which should locate in `C:\Users\your_username\`. If the file does not exist, simply create it. Within the file, copy and paste the setting below. WSL2's VM allocates a certain amount of memory, and Swap space is managed within that VM's memory allocation. This means the Swap memory potentially consumes the RAM in WSL connection. Therefore, the memory needs to be much bigger than Swap to run the sllm successfully. After testing on different WSL config settings, the minimum memory and swap required is listed below.
```

Comment:
Hi,  
Thank you for the modification—it looks great! Could you confirm whether the `sllm-store-server` will fail to run if the WSL settings are configured to less than 22GB of memory and 8GB of swap?

Reply:
Hi, I doublechecked it. It turns out that I didn't clear the cache in previous tests. Now I can confirm 17GB Memory and 8GB Swap is the minimum requirements as I updated in the PR. Thank you.

Here is the test result for the configuration, a Memory of 16GB and a Swap of 8GB.
![af4597fd401ccda39b71e7c7bfd1929](https://github.com/user-attachments/assets/b9aaa967-1e27-4510-bd45-af74e139e5d3)
PS: there was some memory loss while setting the wsl connection.

Reply:
Hi, is this issue related to the `sllm-store-server` not initializing properly, or is it another deployment issue? Were you able to successfully start the `sllm-store-server` with 16GB of memory and 8GB of swap? This error shows a deployment issue, but the store server might be initialized normally.

Reply:
It started this server normally. 4GB memory pool created, server is listen and control loop running, etc. all seem good. But it ran into OOM issue in the deployment. So this should be a deployment issue.

Reply:
Hi, we have confirmed that this issue is caused by an environment dependency related to the hardware information collection feature we recently released, not an OOM error. Could you please re-test the minimum WSL settings after we resolve this issue? From my perspective, the memory allocation for WSL is tied to the size of the `sllm-store-server` memory pool you have configured. By default, it is set to 4GB. Therefore, the minimum requirement should be less than 16GB Memory and 8GB Swap but not lower than 4GB Mem.

Reply:
The tests I've been talking so far are based on the version I installed on Friday by
```bash
# On the head node
conda create -n sllm python=3.10 -y
conda activate sllm
pip install serverless-llm
pip install serverless-llm-store

# On a worker node
conda create -n sllm-worker python=3.10 -y
conda activate sllm-worker
pip install serverless-llm[worker]
pip install serverless-llm-store
```

I
I reinstalled the software today, and the testing results are similar to before. However, I noticed that this warning might be the reason behind the unexpected storage requirements. I'm happy to contribute more in future PRs, but I want to make a phased and conclusive contribution for now if that's okay with you guys.

```vbnet
sllm-serve start
2024-11-25 10:52:33,593 WARNING services.py:407 -- Found multiple active Ray instances: {'172.30.30.103:6379', '0.0.0.0:6379'}. Connecting to latest cluster at 0.0.0.0:6379. You can override this by setting the `--address` flag or `RAY_ADDRESS` environment variable.
2024-11-25 10:52:33,593 INFO worker.py:1540 -- Connecting to existing Ray cluster at address: 0.0.0.0:6379...
2024-11-25 10:52:33,598 INFO worker.py:1724 -- Connected to Ray cluster.
(SllmController pid=2625, ip=172.30.30.103) INFO 11-25 10:52:34 controller.py:60] Starting store manager
(SllmController pid=2625) INFO 11-25 10:52:34 controller.py:73] Starting scheduler
(StoreManager pid=2677) INFO 11-25 10:52:34 store_manager.py:196] Initializing store manager
**(StoreManager pid=2677) WARNING 11-25 10:52:34 store_manager.py:208] Hardware info not provided. Storage-aware scheduling might not work properly**
INFO:     Started server process [2399]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:8343 (Press CTRL+C to quit)
(FcfsScheduler pid=2729) INFO 11-25 10:52:34 fcfs_scheduler.py:53] Starting FCFS scheduler
(FcfsScheduler pid=2729) INFO 11-25 10:52:34 fcfs_scheduler.py:94] Starting control loop
(SllmController pid=2625, ip=172.30.30.103) INFO 11-25 10:52:38 controller.py:101] Registering new model facebook/opt-1.3b
```

Reply:
Hello,  

We’ve identified that the error you encountered is due to an issue with passing the type parameter to the vLLM model downloader. This issue does not occur when using the Hugging Face Transformers backend. Could you please test the minimum setup with the Transformers backend? You can specify this backend when deploying the model:  

```bash
sllm-cli deploy --model facebook/opt-1.3b --backend transformers
```  

Or you can wait until PR #163 is merged and test again.

The reason we ask you to test the minimum configuration is to ensure users can verify if their machines meet the basic requirements to run our service, even on a simple laptop. This is similar to checking the minimum system requirements provided for games on platforms like Steam (if you’re familiar with that). 

Thank you!

Reply:
aha, I see. In that case, the minimum configuration needs 4GB Mem available, which makes the minimum setting in the `.wslconfig` be Memory = 5GB and Swap=2.
```bash
sllm-cli deploy --model facebook/opt-1.3b --backend transformers
```
This does improve the progress and makes the downloading process much faster. Also the configuration of 16GB Memory and 8 Swap that failed to deploy the model using `sllm-cli deploy --model facebook/opt-1.3b` can now deploy the model successfully.

Reply:
Okay, thank you. Can you modify the setting?

Reply:
Hi, you can also see there is still an error in spell check, please add these unrecognized words to wordlist.

Reply:
I was wondering if there is a way to run the github checks locally.